### PR TITLE
CIRC-1331 Add rule for checking ':' before policy

### PR DESCRIPTION
**Pull request by viktor-draban**
## Purpose
The goal is to fix saving rules when they have unfinished criteria ( without ':' and without policies )

[CIRC-1331](https://issues.folio.org/browse/CIRC-1331)

## Approach
Add rule for fallback-policy and add handler for checking stetements

### CODE REVIEW

File: src/main/antlr4/org/folio/circulation/rules/CirculationRules.g4 [hunk 1]
```diff
@@ -147,7 +147,10 @@ criterium : CRITERIUM_LETTER
 
 all : 'all';
 
-fallbackpolicy : 'fallback-policy' policies NEWLINE
+fallbackpolicy : 'fallback-policy'
+                ( policies NEWLINE
+                | NEWLINE { notifyErrorListeners("Policy missing."); }
```
Comment (alexanderkurash) [inline src/main/antlr4/org/folio/circulation/rules/CirculationRules.g4:152]: Do we only need to add this to a fallback policy?

File: src/main/java/org/folio/circulation/rules/Text2Drools.java [hunk 2]
```diff
@@ -158,6 +158,16 @@ private void setIndentation(TerminalNode node) {
     indentation = Integer.parseInt(node.getText().substring(6));
   }
 
+  @Override
+  public void enterSimpleStatement(CirculationRulesParser.SimpleStatementContext ctx) {
+    ExprContext expr = ctx.expr();
+    Token token = ctx.getStart();
+    if (expr.criterium() != null && expr.policies() == null) {
+      throw new org.folio.circulation.rules.CirculationRulesException(
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/rules/Text2Drools.java:166]: Why can't we import this class? Is there another one with the same name already imported?

File: src/main/java/org/folio/circulation/rules/Text2Drools.java [hunk 3]
```diff
@@ -158,6 +158,16 @@ private void setIndentation(TerminalNode node) {
...
+        "Policy missing.", token.getLine(), token.getCharPositionInLine() + 1);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/rules/Text2Drools.java:167]: Which policy is missing? Wouldn't it be clearer to the end user if we just said that ':' symbol is missing?

File: src/main/java/org/folio/circulation/rules/Text2Drools.java [hunk 4]
```diff
@@ -158,6 +158,16 @@ private void setIndentation(TerminalNode node) {
     indentation = Integer.parseInt(node.getText().substring(6));
   }
 
+  @Override
+  public void enterSimpleStatement(CirculationRulesParser.SimpleStatementContext ctx) {
+    ExprContext expr = ctx.expr();
+    Token token = ctx.getStart();
+    if (expr.criterium() != null && expr.policies() == null) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/rules/Text2Drools.java:165]: Is this expression only checks for a missing ':'? What if we create a rule that would have criteria and ':' but no policies, would this also be 'true'? I mean, something like `m book:`

File: src/test/java/org/folio/circulation/rules/Text2DroolsTest.java [hunk 5]
```diff
@@ -482,6 +482,24 @@ void missingLoanPolicy() {
       matches("Policy missing after ':'", 3, 8));
   }
 
+  @Test
+  void missingColonAfterPolicy() {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/circulation/rules/Text2DroolsTest.java:486]: Why is colon missing 'after' a policy? I thought it should go before policies and after criteria.

File: src/main/java/org/folio/circulation/rules/Text2Drools.java (hunk 2 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/rules/Text2Drools.java:166]: yeap, thanks

File: src/main/java/org/folio/circulation/rules/Text2Drools.java (hunk 4 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/rules/Text2Drools.java:165]: This case, you describe, is already handled  on 156 line CirculationRules.g4 and we have a test for that case on 480 line Text2DroolsTest.java So I am sure my code doesn't break this flow.

File: src/main/java/org/folio/circulation/rules/Text2Drools.java (hunk 3 above)
Comment (viktor-draban) [reply src/main/java/org/folio/circulation/rules/Text2Drools.java:167]: I followed my ticket

Expected Results:
Syntax error message 'policy missing'

But of course I don't insist, The message is changed

File: src/test/java/org/folio/circulation/rules/Text2DroolsTest.java (hunk 5 above)
Comment (viktor-draban) [reply src/test/java/org/folio/circulation/rules/Text2DroolsTest.java:486]: thanks, done

File: src/main/antlr4/org/folio/circulation/rules/CirculationRules.g4 (hunk 1 above)
Comment (viktor-draban) [reply src/main/antlr4/org/folio/circulation/rules/CirculationRules.g4:152]: no, but for other cases I added handling in Text2Drools.java

**Review by Mykyta-Varenyk — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**